packages/router/src/router.tsx

```diff
@@ -151,51 +139,112 @@ const Router: React.FC<RouterProps> = ({
   </LocationProvider>
 )
 
-const LocationAwareRouter: React.FC<RouterProps> = ({
+/**
+ * `namedRoutes` is the `routes` object that you import from `@redwoodjs/router`:
+ *
+ * ```js
+ * import { Link, routes } from '@redwoodjs/router'
+ *
+ * // ...
+ *
+ * <Link to={routes.home()}>Home</Link>
+ *
+ * // ...
+ * ```
+ *
+ * It's populated by the router, at run-time.
+ */
+const namedRoutes: AvailableRoutes = {}
+
+const LocationAwareRouter = ({
   useAuth,
   paramTypes,
   pageLoadingDelay,
   children,
-}) => {
+}: React.PropsWithChildren<RouterProps>) => {
   const location = useLocation()
+
   const flatChildArray = flattenAll(children)
 
-  const hasHomeRoute = flatChildArray.some((child) => {
-    if (isRoute(child)) {
-      return child.props.path === '/'
-    }
+  /**
+   * Of the router's children that are routes but aren't the not-found route
```

Comment:
I find this hard to read and understand. Usually when you've written something that just means I don't know how to read it, not that something's actually wrong :)  

But to me I think this would be easier to read if it was something like this "Validate the props of the router children that aren't the not-found route (so routes and redirect routes)"

Reply:
Haha I reread it and it's definitely a mouth full. I think I was trying to capture the nuance that not all the router's children have to be routes. There's thing like Sets too, and even just other things that are technically just invalid that we'll eventually throw out. 

But I just updated it, let me know what you think

Reply:
Much easier to read now 🙂